sphinx builds locally but not via github action

**Describe the bug**
A clear and concise description of what the bug is. If possible, please submit a [Minimal Reproducible Example](https://en.wikipedia.org/wiki/Minimal_reproducible_example).

After starting my repository, `sphinx-build docs/source docs/build` works locally on my machine (in my conda environement) but when pushing I get import errors. For instance:

```
Extension error (sphinx.ext.autosummary):
Handler <function process_generate_options at 0x7fecb6b64cc0> for event 'builder-inited' threw an exception (exception: no module named image_tools.registration.masked_phase_correlation)
Error: Process completed with exit code 2.
``` 
It's not a typo or naming error, any import from my repo (image_tools) fails, as if sphinx was running from an environment before installing the package. Do I need to add steps to configure?

**To Reproduce**
Please enter full details of how to reproduce this bug.

Run the cookiecutter and push to main

**Expected behaviour**
A clear and concise description of what you expected to happen.

Successful build

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -1,3 +1,4 @@
+-e .
 linkify-it-py
 myst-parser
 nbsphinx

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -1,4 +1,4 @@
--e .
+-e . # required to build API docs
 linkify-it-py
 myst-parser
 nbsphinx

--- a/docs/source/_static/css/custom.css
+++ b/docs/source/_static/css/custom.css
@@ -12,7 +12,8 @@ body .bd-article-container {
 max-width: 100em !important;
 }
 /* Custom CSS for the documentation site */
-/* prevent logo from being to wide and text clashing with next item */
+/* prevent logo from being too wide and text clashing with next item */
+/* See https://github.com/pydata/pydata-sphinx-theme/issues/1143#issuecomment-2468763375 */
 
 .navbar-header-items__start .navbar-item {
   width: 100%;

--- a/docs/source/_static/css/custom.css
+++ b/docs/source/_static/css/custom.css
@@ -11,6 +11,29 @@ html[data-theme=light] {
 body .bd-article-container {
 max-width: 100em !important;
 }
+/* Custom CSS for the documentation site */
+/* prevent logo from being to wide and text clashing with next item */
+
+.navbar-header-items__start .navbar-item {
+  width: 100%;
+}
+
+.navbar-item .navbar-brand {
+  width: 100%;
+}
+
+.navbar-brand img {
+  min-width: 0;
+  height: auto;
+  max-height: 100%;
+  flex-shrink: 1;
+}
+
+.navbar-brand p {
+  flex: 0 1 auto;
+}
+
+/* sponsors */
 
 .col {
 flex: 0 0 50%;

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -1,13 +1,14 @@
 # conf.py
 # Configuration file for the Sphinx documentation builder.
-import setuptools_scm
+from importlib.metadata import version as get_version
 
 
 project = "python-cookiecutter"
 copyright = "2025, University College London"
 author = "Neuroinformatics Unit"
+
 try:
-    full_version = setuptools_scm.get_version(root="../..", relative_to=__file__)
+    full_version = get_version(project)
     # Splitting the release on '+' to remove the commit hash
     release = full_version.split('+', 1)[0]
 except LookupError:
@@ -84,6 +85,7 @@ html_theme_options = {
         },
     ],
     "logo": {
+        # The release version can be included here if desired later by using f"{project} {release}"
         "text": f"{project}",
     },
     "footer_start": ["footer_start"],

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -85,8 +85,7 @@ html_theme_options = {
         },
     ],
     "logo": {
-        # The release version can be included here if desired later by using f"{project} {release}"
-        "text": f"{project} {release}",
+        "text": f"{project} v{release}",
     },
     "footer_start": ["footer_start"],
     "footer_end": ["footer_end"],

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -86,7 +86,7 @@ html_theme_options = {
     ],
     "logo": {
         # The release version can be included here if desired later by using f"{project} {release}"
-        "text": f"{project}",
+        "text": f"{project} {release}",
     },
     "footer_start": ["footer_start"],
     "footer_end": ["footer_end"],

--- a/docs/source/contributing.md
+++ b/docs/source/contributing.md
@@ -53,6 +53,7 @@ If you are contributing to the Documentation, Before pushing your changes, you c
     the docs:
 
     ``` sh
+     # Must execute from project root directory
     pip install -r docs/requirements.txt
     sphinx-build docs/source docs/build
     ```
@@ -61,7 +62,7 @@ Alternatively, you can use the following commands to install the
 dependencies and build the docs:
 
  ``` sh
-pip install -r docs/requirements.txt
+pip install -r docs/requirements.txt  # Must execute this command from project root directory
 cd docs
 make html
 ```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,3 +1,22 @@
+[project]
+name = "python-cookiecutter"
+authors = [{ name = "Neuroinformatics Unit" }]
+description = "A tool to automatically create a Python project structure ready to release via GitHub and PyPI."
+readme = "README.md"
+dynamic = ["version"]
+
+license = { text = "BSD-3-Clause" }
+
+[build-system]
+requires = [
+    "setuptools>=64",
+    "wheel",
+    "setuptools-scm[toml]>=8",
+]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+
 [tool.ruff]
 line-length = 79
 exclude = ["__init__.py", "build", ".eggs"]

--- a/{{cookiecutter.package_name}}/docs/requirements.txt
+++ b/{{cookiecutter.package_name}}/docs/requirements.txt
@@ -1,3 +1,4 @@
+-e .
 linkify-it-py
 myst-parser
 nbsphinx

--- a/{{cookiecutter.package_name}}/docs/requirements.txt
+++ b/{{cookiecutter.package_name}}/docs/requirements.txt
@@ -1,4 +1,4 @@
--e .
+-e . # required to build API docs
 linkify-it-py
 myst-parser
 nbsphinx

--- a/{{cookiecutter.package_name}}/docs/source/conf.py
+++ b/{{cookiecutter.package_name}}/docs/source/conf.py
@@ -9,7 +9,7 @@
 import os
 import sys
 
-import setuptools_scm
+from importlib.metadata import version as get_version
 
 # Used when building API docs, put the dependencies
 # of any class you are documenting here
@@ -24,7 +24,9 @@ project = "{{cookiecutter.package_name}}"
 copyright = "2022, {{cookiecutter.full_name}}"
 author = "{{cookiecutter.full_name}}"
 try:
-    release = setuptools_scm.get_version(root="../..", relative_to=__file__)
+    full_version = get_version(project)
+    # Splitting the release on '+' to remove the commit hash
+    release = full_version.split('+', 1)[0]
 except LookupError:
     # if git is not initialised, still allow local build
     # with a dummy version

